examples/qm7/qm7_ANI.py

```diff
@@ -10,51 +10,83 @@
 import tensorflow as tf
 tf.set_random_seed(123)
 import deepchem as dc
+import os
+
+HARTREE_TO_KCAL_PER_MOL = 627.509
 
-# Load Tox21 dataset
 tasks, datasets, transformers = dc.molnet.load_qm7_from_mat(
-    featurizer='BPSymmetryFunction')
-train_dataset, valid_dataset, test_dataset = datasets
+    featurizer='BPSymmetryFunction', split='index', reload=False)
+all_dataset = dc.data.DiskDataset.merge(datasets)
+invalid_inds = []
+X = all_dataset.X
+for i in range(X.shape[0]):
+  # Exclude all molecules having S
+  if 16 in X[i, :, 0]:
+    invalid_inds.append(i)
+valid_inds = np.delete(np.arange(all_dataset.y.shape[0]), invalid_inds)
+dataset = all_dataset.select(valid_inds)
+
+splitter = dc.splits.RandomSplitter()
+train, valid, test = splitter.train_valid_test_split(dataset)
+
+y = dc.trans.undo_transforms(train.y, transformers) / HARTREE_TO_KCAL_PER_MOL
+train = dc.data.DiskDataset.from_numpy(
+    train.X, y, w=train.w, ids=train.ids, tasks=train.tasks)
+
+y = dc.trans.undo_transforms(valid.y, transformers) / HARTREE_TO_KCAL_PER_MOL
+valid = dc.data.DiskDataset.from_numpy(
+    valid.X, y, w=valid.w, ids=valid.ids, tasks=valid.tasks)
+
+y = dc.trans.undo_transforms(test.y, transformers) / HARTREE_TO_KCAL_PER_MOL
+test = dc.data.DiskDataset.from_numpy(
+    test.X, y, w=test.w, ids=test.ids, tasks=test.tasks)
 
 # Batch size of models
 max_atoms = 23
 batch_size = 128
-layer_structures = [128, 128, 64]
-atom_number_cases = [1, 6, 7, 8, 16]
-
-ANItransformer = dc.trans.ANITransformer(
-    max_atoms=max_atoms, atom_cases=atom_number_cases)
-train_dataset = ANItransformer.transform(train_dataset)
-valid_dataset = ANItransformer.transform(valid_dataset)
-test_dataset = ANItransformer.transform(test_dataset)
-n_feat = ANItransformer.get_num_feats() - 1
+layer_structures = [64, 64, 32]
+atom_number_cases = [1, 6, 7, 8]
 
 # Fit models
 metric = [
     dc.metrics.Metric(dc.metrics.mean_absolute_error, mode="regression"),
     dc.metrics.Metric(dc.metrics.pearson_r2_score, mode="regression")
 ]
 
-model = dc.models.ANIRegression(
-    len(tasks),
-    max_atoms,
-    n_feat,
-    layer_structures=layer_structures,
-    atom_number_cases=atom_number_cases,
-    batch_size=batch_size,
-    learning_rate=0.001,
-    use_queue=False,
-    mode="regression")
-
-# Fit trained model
-model.fit(train_dataset, nb_epoch=300, checkpoint_interval=100)
-
-print("Evaluating model")
-train_scores = model.evaluate(train_dataset, metric, transformers)
-valid_scores = model.evaluate(valid_dataset, metric, transformers)
-
-print("Train scores")
-print(train_scores)
-
-print("Validation scores")
-print(valid_scores)
+model_dir = '/home/zqwu/deepchem/examples/qm7/ANI1_model'
```

Comment:
Can you remove the hardcoded directory name?

Reply:
Ah sorry, forget to clean up this part